### PR TITLE
Health check bugfix

### PR DESCRIPTION
**Pull request by Eriken79**
This pull request fixes some errors around the Health Check mechanism. The previous version of this mechanism hard coded the max_tokens variable that is packaged with the request to the model at 1. This was fine for previous models to check their availability, but newer models take more varied inputs to that field and need higher max_tokens. These changes generalize the request sent by the health check mechanism to be more generically applicable to the needs of more modern models. max_tokens has been raised to 4096 which works generally for a lot of newer models and an exception has been made for models that receive max_completion_tokens. This mechanism is more extensible in case other corner cases are encountered. Also fixed a small issue with test connection failing on Ollama since it didn't have auth embedded in the request.

### CODE REVIEW

File: src/api_health.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    rejected = re.search(
+        rf"(?:unsupported|unrecognized|unknown) (?:parameter|argument|field)[: ]*['\"]?{field}\b"
+        rf"|\b{field}\b['\"]? (?:is |are )?(?:not supported|unsupported|not allowed)",
+        detail,
+    )
+    if rejected:
+        kwargs["json"] = dict(kwargs["json"])
+        kwargs["json"]["max_tokens"] = kwargs["json"].pop(field)
+        response = requests.post(**kwargs)
+    return response
```
Comment (jdunns) [inline src/api_health.py:57]: Seems pretty straightforward.

File: src/custom_api_client.py [hunk 2]
```diff
@@ -743,22 +745,17 @@ def do_post():
...
-            else:
-                api_url = f"{self.base_url}/v1/chat/completions"
-
-            request_kwargs = {"headers": headers, "json": test_payload, "timeout": 5}
-            if not self.verify_ssl:
-                request_kwargs["verify"] = False
-
-            response = requests.post(api_url, **request_kwargs)
+            request = build_health_request(self.base_url, self.api_key, self.default_model, self.verify_ssl)
+            response = post_health_request(request)
```
Comment (jdunns) [inline src/custom_api_client.py:752]: Does this timeout if the endpoint isn't available?

File: tests/test_api_health.py
Comment (jdunns) [inline tests/test_api_health.py:1]: And you wrote tests. Fantastic.

**Review by jdunns — APPROVED** [review]

File: src/custom_api_client.py (hunk 2 above)
Comment (Eriken79) [reply src/custom_api_client.py:752]: yes, in src/api_health.py on line 24 it sets the timeout, i have it set at 10 seconds to establish a connection and 30 to receive data, which is maybe a bit long but it can be adjusted whenever
